docs/examples/physics_using_pybullet/viz_domino.py

```diff
@@ -0,0 +1,238 @@
+* Calculate the Rotation Matrix.
+  * Get the difference in orientations (Quaternion).
+  * Generate the corresponding rotation matrix according to that difference.
+  * Reshape it in a 3x3 matrix.
+* Perform calculations to get the required position and orientation.
+* Update the position and orientation.
+"""
+
+def sync_brick(object_index, multibody):
+    pos, orn = p.getBasePositionAndOrientation(multibody)
+
+    rot_mat = np.reshape(
+        p.getMatrixFromQuaternion(
+            p.getDifferenceQuaternion(orn, brick_orns[object_index])),
+        (3, 3))
+
+    vertices[object_index * sec: object_index * sec + sec] = \
+        (vertices[object_index * sec: object_index * sec + sec] -
+         brick_centers[object_index]) @ rot_mat + pos
+
+    brick_centers[object_index] = pos
+    brick_orns[object_index] = orn
+
+
+###############################################################################
+# A simpler but inaccurate approach is used here to update the position and
+# orientation.
+
+
+def sync_actor(actor, multibody):
+    pos, orn = p.getBasePositionAndOrientation(multibody)
+    actor.SetPosition(*pos)
+    orn_deg = np.degrees(p.getEulerFromQuaternion(orn))
+    actor.SetOrientation(*orn_deg)
+
+
+###############################################################################
+# Here, we define a textblock to display the Avg. FPS and simulation steps.
+
+fpss = np.array([])
+tb = ui.TextBlock2D(text="Avg. FPS: \nSim Steps: ", position=(0, 680),
+                    font_size=30, color=(1, 0.5, 0))
+scene.add(tb)
+
+###############################################################################
+# Set the camera for better visualization.
+
+scene.set_camera(position=(10.46, -8.13, 6.18),
+                 focal_point=(0.0, 0.0, 0.79),
+                 view_up=(-0.27, 0.26, 0.90))
+
+
+###############################################################################
+# Timer callback is created which is responsible for calling the sync and
+# simulation methods.
+
+
+# Create timer callback which will execute at each step of simulation.
+def timer_callback(_obj, _event):
+    global apply_force, fpss
+    cnt = next(counter)
+    showm.render()
+
+    if cnt % 1 == 0:
+        fps = scene.frame_rate
+        fpss = np.append(fpss, fps)
+        tb.message = "Avg. FPS: " + str(np.round(np.mean(fpss), 0)) + \
+                     "\nSim Steps: " + str(cnt)
+
+    # Get the position and orientation of the first brick.
+    brick1_pos, brick1_orn = p.getBasePositionAndOrientation(bricks[0])
+
+    # Apply force for the first step of simulation.
+    if apply_force:
+        # Apply the force.
+        p.applyExternalForce(bricks[0], -1,
+                             forceObj=[5, 0, 0],
+                             posObj=brick1_pos,
+                             flags=p.WORLD_FRAME)
+        apply_force = False
+
+    # Set position and orientation of the first Brick.
+    sync_actor(brick_actor, bricks[0])
```

Comment:
All your dominoes are rendered by this single actor called `brick_actor`. So whatever modifications you make to this actor will be reflected on all the dominoes.
Thus, in this case when you pass `brick_actor`, `bricks` to `sync_actor` you are basically asking fury to modify all the dominoes with the position and orientation of the first domino, which is incorrect.

That's why you can observe that all of your dominoes are following the same trajectory.